snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedMrcmRestService.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2022 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.core.rest;
+
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.web.bind.annotation.*;
+
+import com.b2international.snowowl.core.events.util.Promise;
+import com.b2international.snowowl.core.rest.AbstractRestService;
+import com.b2international.snowowl.snomed.core.domain.refset.SnomedReferenceSetMembers;
+import com.b2international.snowowl.snomed.datastore.request.MrcmTypeRequest.ATTRIBUTE_TYPE;
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+/**
+ * @since 8.7.0
+ */
+@Tag(description="Rules", name = "rules")
+@RestController
+public class SnomedMrcmRestService extends AbstractRestService {
+	
+	@Operation(
+		summary="Retrieve MRCM attribute domain type refset members filtered by domain and attribute type", 
+		description="Retrieve MRCM relationship type rules ")
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "400", description = "Bad Request"),
+		@ApiResponse(responseCode = "404", description = "Not found"),
+	})
+	@GetMapping(value = "/{path:**}/types", produces = { AbstractRestService.JSON_MEDIA_TYPE })
```

Comment:
Not sure about the endpoint's name, I'd like to see the word mrcm in it. `mrcm-rules` perhaps, but I'm not a fan of dashes in endpoint names. Another option would be `{path:**}/mrcm/types`. @nagyo @apeteri thoughts?

Reply:
"Attributes" and "attribute ranges" should be incorporated into each path in some form, this is how the documentation refers to relationship types in this context.